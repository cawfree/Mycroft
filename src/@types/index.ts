import {Browser} from 'puppeteer-core';

export type BrowserContext = Awaited<ReturnType<Browser['createIncognitoBrowserContext']>>;

export type StealthBrowserContext = {
  readonly browserContext: BrowserContext;
  readonly close: () => Promise<void>;
};

export type Contest = {
  readonly contestPageUrl: string;
  readonly issueRepoUrl: string;
};

export type GetContestsParams = {
  readonly watson: string;
};

export type GetContestsResult = {
  /// @dev The contests a Watson has participated in.
  readonly contests: readonly Contest[];
};

export type GetContests = (params: GetContestsParams) => Promise<GetContestsResult>;

export type Mycroft = Omit<StealthBrowserContext, 'browserContext'> & {
  readonly getContests: GetContests;
};

