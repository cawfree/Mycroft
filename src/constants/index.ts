import Bottleneck from 'bottleneck';

// The maximum number of parallel requests to use
// when querying Sherlock. We ❤️ Sherlock, so we'll
// keep this low.
export const BOTTLENECK_SHERLOCK = new Bottleneck({
  maxConcurrent: 5,
});

