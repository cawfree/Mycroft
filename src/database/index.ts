export * from './getCollection';
export * from './getRandomFinding';

