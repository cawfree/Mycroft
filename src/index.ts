export * from './@types';
export * from './constants';
export * from './mycroft';
export * from './puppeteer';

