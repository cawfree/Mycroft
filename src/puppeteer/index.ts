export * from './createStealthBrowserContext';
export * from './getAllPageLinks';
export * from './openPageAndWaitUntilLoaded';
