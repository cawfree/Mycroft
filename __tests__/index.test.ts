import 'jest';

import {createMycroft} from '../src';

jest.setTimeout(10 * 60 * 1_000);

const createTestMycroft = () => createMycroft({
  headless: false,
});

describe('jest', () => {

  it('get contest urls', async () => {
    
    const {close, getContests} = await createTestMycroft();

    /// @dev Note, this will get invalidated as the user participates
    ///      in more contests. You'll probably need to update snapshots
    //       frequently with a call to `yarn test -u`.
    expect(await getContests({watson: 'cawfree'})).toMatchSnapshot();

    await close();

  });

});

