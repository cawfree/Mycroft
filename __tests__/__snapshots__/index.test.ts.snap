// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest get contest urls 1`] = `
{
  "contests": [
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/111",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-09-nounsbuilder-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/135",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-12-dodo-gsp-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/123",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-10-perennial-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/122",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-10-looksrare-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/120",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-10-aloe-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/114",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-09-ajna-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/109",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-09-Gitcoin-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/101",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-06-tokemak-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/97",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-06-gfx-judging/issues",
    },
    {
      "contestPageUrl": "https://audits.sherlock.xyz/contests/86",
      "issueRepoUrl": "https://github.com/sherlock-audit/2023-06-arrakis-judging/issues",
    },
  ],
}
`;
